core/src/main/java/org/apache/accumulo/core/classloader/ClassLoaderUtil.java

```diff
@@ -36,25 +39,30 @@ private ClassLoaderUtil() {
   /**
    * Initialize the ContextClassLoaderFactory
    */
-  public static synchronized void initContextFactory(AccumuloConfiguration conf) {
+  public static synchronized void initContextFactory(final AccumuloConfiguration conf) {
     if (FACTORY == null) {
       LOG.debug("Creating {}", ContextClassLoaderFactory.class.getName());
       String factoryName = conf.get(Property.GENERAL_CONTEXT_CLASSLOADER_FACTORY);
       if (factoryName == null || factoryName.isEmpty()) {
-        // load the default implementation
         LOG.info("Using default {}, which is subject to change in a future release",
             ContextClassLoaderFactory.class.getName());
-        FACTORY = new DefaultContextClassLoaderFactory(conf);
-      } else {
-        // load user's selected implementation
-        try {
-          var factoryClass = Class.forName(factoryName).asSubclass(ContextClassLoaderFactory.class);
-          LOG.info("Creating {}: {}", ContextClassLoaderFactory.class.getName(), factoryName);
-          FACTORY = factoryClass.getDeclaredConstructor().newInstance();
-        } catch (ReflectiveOperationException e) {
-          throw new IllegalStateException("Unable to load and initialize class: " + factoryName, e);
-        }
+        factoryName = DefaultContextClassLoaderFactory.class.getName();
```

Comment:
The default implementation is intentionally treated differently, so users cannot configure it or instantiate it directly. It's for internal use only.

Reply:
hmm... ok

Reply:
Are you saying that I need to make this such that the user cannot set DefaultContextClassLoaderFactory as the value for this property

Reply:
Adding a check to ensure DefaultContextClassLoaderFactory is not set as the property name.  IMHO I think this is not reasonable.  If you are going to supply a default implementation, then let that be part of the external API.

Reply:
ok